Examples/Rendering/LabelmapMIP/index.js

```diff
@@ -0,0 +1,260 @@
+  const numberOfComponents = 1;
+
+  createSphericalLabel(
+    labelmapArray,
+    dims,
+    center,
+    radius1,
+    1,
+    numberOfComponents
+  );
+  createSphericalLabel(
+    labelmapArray,
+    dims,
+    [center[0] + 2 * radius1, center[1] + 2 * radius1, center[2] + 2 * radius1],
+    radius2,
+    2,
+    numberOfComponents
+  );
+
+  const labelmapActor = vtkVolume.newInstance();
+  const labelmapMapper = vtkVolumeMapper.newInstance();
+
+  // Labelmap pipeline
+  labelmapMapper.setInputData(labelMapData);
+  labelmapActor.setMapper(labelmapMapper);
+
+  const labelMapColorFunction = vtkColorTransferFunction.newInstance();
+  labelMapColorFunction.addRGBPoint(0, 0, 0, 0);
+  labelMapColorFunction.addRGBPoint(1, 0, 0, 1); // Blue for label 1
+  labelMapColorFunction.addRGBPoint(2, 1, 0, 0); // Red for label 2
+
+  const labelMapOpacityFunction = vtkPiecewiseFunction.newInstance();
+  labelMapOpacityFunction.addPoint(0, 0);
+  labelMapOpacityFunction.addPoint(1, 0.5);
+  labelMapOpacityFunction.addPoint(2, 0.5);
+
+  labelmapActor.getProperty().setRGBTransferFunction(0, labelMapColorFunction);
+  labelmapActor.getProperty().setScalarOpacity(0, labelMapOpacityFunction);
+
+  // set interpolation to nearest to avoid blending
+  labelmapActor.getProperty().setInterpolationTypeToNearest();
+  renderer.addVolume(labelmapActor);
+}
+
+function createAdvancedMIPLabelmap(imageData, dims, center, radius1, radius2) {
+  const array = imageData.getPointData().getArray(0);
+  const baseData = array.getData();
+
+  const numberOfComponents = 2;
+  const cubeData = new Float32Array(numberOfComponents * baseData.length);
+
+  // Copy original data to first component
+  for (let i = 0; i < baseData.length; i++) {
+    cubeData[i * numberOfComponents] = baseData[i];
+  }
+
+  createSphericalLabel(cubeData, dims, center, radius1, 1, numberOfComponents);
+  createSphericalLabel(
+    cubeData,
+    dims,
+    [center[0] + 2 * radius1, center[1] + 2 * radius1, center[2] + 2 * radius1],
+    radius2,
+    2,
+    numberOfComponents
+  );
+
+  actor.getProperty().setColorMixPreset(1);
+
+  const maskCtfun = vtkColorTransferFunction.newInstance();
+  maskCtfun.addRGBPoint(0, 0, 0, 0);
+  maskCtfun.addRGBPoint(1, 0, 0, 1); // blue  to red
+  maskCtfun.addRGBPoint(2, 1, 0, 0); // red to green
+
+  const maskOfun = vtkPiecewiseFunction.newInstance();
+  maskOfun.addPoint(0, 0);
+  maskOfun.addPoint(1, 1);
+  maskOfun.addPoint(2, 1);
+
+  const arrayAgain = mapper.getInputData().getPointData().getArray(0);
+  arrayAgain.setData(cubeData);
+  arrayAgain.setNumberOfComponents(2);
+
+  actor.getProperty().setRGBTransferFunction(1, maskCtfun);
+  actor.getProperty().setScalarOpacity(1, maskOfun);
+  actor.getProperty().setForceNearestInterpolation(1, true);
+
+  // same thickness
+  actor.getProperty().setLabelOutlineThickness(3);
+  // per label thickness
+  //   actor.getProperty().setLabelOutlineThickness([5, 1]);
```

Comment:
Added

Reply:
Thank you!
This works well when switching once from regular MIP to labelmap MIP and back to regular MIP!
But if you switch several times, the volume will break a little bit more, every time you switch to labelmap MIP
Can you also remove this commented code?

Reply:
I see that you use a controlPanel string instead of a controller.html file as done in the other examples
Did you take this idea from the vtk_vanilla.md file ? This is an example file for using vtk.js but not what is done in the vtk.js examples
Can you move it to a controller.html file and import it using import controlPanel from './controller.html';?